### PR TITLE
Firefox fixes

### PR DESCRIPTION
**Pull request by marusak**
This fixes 3 out of 7 or 8 broken tests. The last commit is rather intrusive so I want to see if it actually works or if it has some issues. It many actually fix some of those remaining tests as well.

### CODE REVIEW

File: test/common/firefox-cdp-driver.js [hunk 1]
```diff
@@ -91,22 +91,25 @@ function setupLogging(client) {
 
     function processException(info) {
         let details = info.exceptionDetails;
+        if (details.exception)
+            details = details.exception;
+
         // don't log test timeouts, they already get handled
-        if (details.exception && details.exception.className === "PhWaitCondTimeout")
+        if (details.className === "PhWaitCondTimeout")
```
Comment (martinpitt) [inline test/common/firefox-cdp-driver.js:97]: Note that our current tasks container use nightly firefox (86.0a1 right now). I'd really like this to work with the current release (84) as well -- did you try this locally? I assume you did, but I'd like to clarify if at least for some time we'll need both code paths.

File: test/common/cdp.py [hunk 2]
```diff
@@ -221,7 +221,7 @@ def start(self):
             self._browser_home = tempfile.mkdtemp()
             environ = os.environ.copy()
             environ["HOME"] = self._browser_home
-            environ["LC_ALL"] = "C.utf8"
+            environ["LC_ALL"] = "en_US.UTF-8"
```
Comment (martinpitt) [inline test/common/cdp.py:224]: I'm afraid this won't work in our current tasks containers just yet, as they don't have any locales defined:
```
❱❱❱ locale -a
C
C.utf8
POSIX
```
If that is the right workaround, then we need to install `glibc-langpack-en` into the tasks container.

**Review by martinpitt — COMMENTED** [review]
Nice progress! BTW, ignore the unit-tests failures for now. TL/DR, I'm on it.

File: test/common/firefox-cdp-driver.js (hunk 1 above)
Comment (martinpitt) [reply test/common/firefox-cdp-driver.js:97]: Oh, silly me -- it's right there in line 94.

Comment (martinpitt) [reply test/common/firefox-cdp-driver.js:97]: Good point, I forgot that we *have* to use nightly due to releases not having the devtools driver.

File: test/common/cdp.py (hunk 2 above)
Comment (martinpitt) [reply test/common/cdp.py:224]: FTR: I tested that in my toolbox (which *is* the cockpit/tasks container) with `LC_ALL=en_US.UTF-8 HOME=/tmp/f firefox`. I get a lot of errors on the console like
```
(firefox:237002): Gtk-WARNING **: 17:13:07.810: Locale not supported by C library.
	Using the fallback 'C' locale.
```
but the string comparison returns -1 in the developer console.

**Review by martinpitt — DISMISSED** [review]
Super-nice cleanup! Even if there are some leftover failures (but no regressions), feel free to land the current lot.

**Comment by martinpitt** [comment]
Firefox seems to have a weird string collation in general then. Most languages/dictionaries and also ASCII sort upper-case before lower case. E.g. compare with
```
❱❱❱ LC_ALL=C.UTF-8 python3 -c 'print("Virtio block device" < "Virtio SCSI")'
False

❱❱❱ LC_ALL=en_US.UTF-8 python3 -c 'print("Virtio block device" < "Virtio SCSI")'
False
```

Same results with comparing `a` with `A`, which is a bit clearer :-)

**Comment by marusak** [comment]
Machines tests still broken, but no regression.

**Comment by marusak** [comment]
One failed test, seems unrelated, lets retry. [log](https://logs.cockpit-project.org/logs/pull-15097-20210105-190418-b9ffb65d-fedora-32-firefox/log.html#293)

File: test/verify/check-machines [hunk 3]
```diff
@@ -2532,7 +2532,8 @@ class TestMachines(MachineCase, StorageHelpers, NetworkHelpers):
                 else:
                     raise AssertionError("Unknown disk device")
             else:
-                b.wait_in_text("#vm-{0}-disks .ct-table-empty".format(name), "No disks defined")
+                # On smaller resolutions it is not visible, but always in DOM
+                b.wait_present("#vm-{0}-disks .ct-table-empty:contains('No disks defined')".format(name))
```
Comment (martinpitt) [inline test/verify/check-machines:2536]: Hard to say without a screenshot, but could we make this a `# HACK` and conditional to firefox?

Comment (martinpitt) [reply test/verify/check-machines:2536]: Also, judging by the amount of hacks we had to do in selenium/edge tests (which does have that weird "scrolled off == !visible" behaviour), we should run into this in more cases if it was generally an issue.

Comment (marusak) [reply test/verify/check-machines]: This is not that kind of hack, that it is visible, just the protocol does not think so. It literally gets hidden when you make your window smaller. (Just go to VM details that has no disks/networks and make your window smaller and bigger).

I changed it to test it being visible on chrome and not visible (but present) on firefox. This way we can check it works in both ways. I don't call it hack for that reason.

**Review by martinpitt — APPROVED** [review]
Nevermind, *now* I understand it -- we literally run Firefox with a smaller resolution. The comment cleared it up. As I'm AFK over the day, approving (assuming there are no test regressions). Many thanks! :clap:

**Comment by martinpitt** [comment]
Also, this PR should totally have been called "Firefixes"!

**Comment by martinpitt** [comment]
@marusak :green_heart: nice work!

**Pull request merged**